### ISSUE TITLE
enable hyphenation based on wiki language

### ISSUE THREAD

**Issue by Juergen-aus-Zuendorf** [splitbrain/dokuwiki-plugin-dw2pdf#415]
Hi,

it seems that the given language is not passed from HTML to the PDF output. I am trying to use automatic hyphenation using the CSS function "hyphens". But for this to work, the definition of the language attribute in the HTML code is required. When debugging with "&debughtml=text" you can see that the language is not specified.

Regards
Juergen

**Comment by Klap-in** [comment]
Could you describe how someone can reproduce your situation? I have not yet an idea what your issue is.

**Comment by Juergen-aus-Zuendorf** [comment]
The CSS command `Hyphens: auto` can be used to activate automatic hyphenation. However, the hyphenation rules are language-specific. In HTML, the language is determined by the `lang` attribute, and browsers hyphenate only if this attribute is present and the corresponding hyphenation dictionary is available. Interestingly, mpdf also supports this function. But without a specific, previously defined language, hyphenation looks very strange.

**Comment by Klap-in** [comment]
And I can just export every page to see this? Please provide a small recipe for reproducing. I do not know what I should put in my wiki, which plugins/versions/settings are needed. I don't know what the current output is in your setup. I do not know what output you expect. Please realize that we do not know anything about your setup. So guide your potential helpers to quickly reproduce your situation.

**Comment by Juergen-aus-Zuendorf** [comment]
Here I have a small recipe as a wiki text:
```
====== Spielplatz ======

Die Betriebsanleitung vor Beginn aller Arbeiten sorgfältig durchlesen! Sie ist Produktbestandteil und muss in unmittelbarer Nähe der Maschine für das Personal jederzeit zugänglich aufbewahrt werden. 
```
And here is the result:
a) Browser (Edge):
![Browser(Edge)](https://user-images.githubusercontent.com/1800472/97729041-8d3d6f00-1ad2-11eb-9687-ce5a52381cca.png)

b) Output as PDF:
![PDF-Ausgabe](https://user-images.githubusercontent.com/1800472/97730144-c9bd9a80-1ad3-11eb-949b-1b0864fbf269.png)

The word "jederzeit" is hyphened at a wrong position (correct should be this: je-der-zeit). And this problem runs through the whole text.

My idea why the hyphenation is in the wrong position is that the language "de" is not recognized because it is simply not known during PDF generation. As written at the beginning of this thread, when debugging with "&debughtml=text" you can see that the language is not specified.

Here is my template:
[schuette_ban_de_v01.zip](https://github.com/splitbrain/dokuwiki-plugin-dw2pdf/files/5466270/schuette_ban_de_v01.zip)

**Comment by Juergen-aus-Zuendorf** [comment]
I have done a little research:
The language must be defined in the file 
"..\dw2pdf\vendor\mpdf\mpdf\src\Config\ConfigVariables.php"

Syntax for German hyphenation:  
`'SHYlang' => "de", // 'en','de','es','fi','fr','it','nl','pl','ru','sv'`

**Comment by Klap-in** [comment]
The wiki has already a global language setting. (of course it get more complex to get the correct language when the translation plugin is used. https://www.dokuwiki.org/plugin:translation). 

From this setting, probably for available languages in mpdf the hyphenation setting could be set.
https://mpdf.github.io/what-else-can-i-do/hyphenation.html

The config settings for mpdf are handled here: https://github.com/splitbrain/dokuwiki-plugin-dw2pdf/blob/master/DokuPDF.class.php

**Comment by Juergen-aus-Zuendorf** [comment]
In fact, I use the translation plugin. And for this reason, unfortunately, the global definition via the wiki login language is not enough for me.
Therefore, it would certainly be ideal to determine the language analogous to the translation plugin. Alternatively, it would be sufficient to have an URL parameter, e.g. "&lang=en".

**Comment by splitbrain** [comment]
implemented in #481
